libnest/src/cache/depgraph/requirement.rs

```diff
@@ -38,40 +38,64 @@ impl std::fmt::Display for RequirementKind {
 #[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
 pub struct Requirement {
     kind: RequirementKind,
-    fulfilling: Option<NodeID>,
+    management_method: RequirementManagementMethod,
     fulfilled: NodeID,
+    fulfilling: Option<NodeID>,
 }
 
 impl Requirement {
     #[allow(dead_code)] // TODO: Remove this when the function is used
     #[inline]
     pub(crate) fn from(
         kind: RequirementKind,
-        fulfilling: Option<NodeID>,
+        management_method: RequirementManagementMethod,
         fulfilled: NodeID,
     ) -> Requirement {
         Requirement {
             kind,
-            fulfilling,
+            management_method,
             fulfilled,
+            fulfilling: None,
         }
     }
 
-    /// Returns a reference to the kind of this requirement.
+    /// Returns a reference to the kind of this requirement
     #[inline]
     pub fn kind(&self) -> &RequirementKind {
         &self.kind
     }
 
-    /// Returns the [`NodeID`] of the [`Node`] that fulfills this requirement.
+    /// Returns the requirement method for this requirement
+    #[inline]
+    pub fn management_method(&self) -> RequirementManagementMethod {
+        self.management_method
+    }
+
+    /// Returns a reference to the [`NodeID`] of the [`Node`] that fulfills this requirement
     #[inline]
-    pub fn fulfilling_node(&self) -> Option<NodeID> {
-        self.fulfilling
+    pub fn fulfilling_node_id(&self) -> &Option<NodeID> {
```

Comment:
Why returning by reference instead of returning by copy? No big deal here, but it looks like you did it on purpose and i'm genuinely curious.

Reply:
It was only to match the `_mut` getter, which itself returns a reference.